provisioning/device/src/interfaces.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+import { SharedAccessSignature, X509 } from 'azure-iot-common';
+
+/**
+ * @private
+ * Callback for responses from the previsioning service
+ */
+export type ResponseCallback = (err?: Error, response?: any) => void;
+
+/**
+ * type defining the types of authentication we support
+ */
+export type Authentication = string | X509 | SharedAccessSignature;
+
+export interface ClientConfiguration {
+  /**
+   * User-Agent string passed to the service as part of communication
+   */
+  userAgent: string;
+
+  /**
+   * Default interval for polling, to use in case service doesn't provide it to us.
+   */
+  pollingInterval: number;
+
+  /**
+   * Default host for the provisioning service
+   */
+  provisioningHost: string;
+
+  /**
+   * apiVersion to use while communicating with service.
+   */
+  apiVersion: string;
+
+  /**
+   * default timeout to use when communicating with the service
+   */
+  timeoutInterval: number;
+
+  /**
+   * idScope to use when communicating with the provisioning service
+   */
+  idScope: string;
```

Comment:
>idScope [](start = 2, length = 7)

public API nitpicking: I believe other SDKs use scopeId.

Reply:
nope.  csharp uses idScope, which matches the portal UI

---
In reply to: [148917686](https://github.com/Azure/azure-iot-sdk-node/pull/135#discussion_r148917686) [](ancestors = 148917686)